src/main/java/com/example/bankservicedemo/service/impl/TransactionServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.example.bankservicedemo.service.impl;
+
+import com.example.bankservicedemo.exception.NoEnoughMoneyException;
+import com.example.bankservicedemo.model.Account;
+import com.example.bankservicedemo.model.Transaction;
+import com.example.bankservicedemo.repository.TransactionRepository;
+import com.example.bankservicedemo.service.AccountService;
+import com.example.bankservicedemo.service.TransactionService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class TransactionServiceImpl implements TransactionService {
+    private final TransactionRepository transactionRepository;
+    private final AccountService accountService;
+
+    @Override
```

Comment:
let's use the `@Transactional` annotation to make this whole logic in this method be executed in a single transaction
see this article https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/annotation/Transactional.html